dask_expr/_collection.py

```diff
@@ -283,7 +280,18 @@ def optimize(self, fuse: bool = True):
 
     @property
     def dask(self):
-        return self.__dask_graph__()
+        # FIXME: This is highly problematic. Defining this as a property can
+        # cause very unfortunate materializations. Even a mere hasattr(obj,
+        # "dask") check already triggers this since it's a property, not even a
+        # method.
+        return self.__dask_graph_factory__().optimize().materialize()
+
+    def finalize_compute(self):
+        return new_collection(Repartition(self.expr, 1))
```

Comment:
This is very problematic in the _fairly common_ use case where the client mounts a lot more memory than a single worker. This forces the whole object to be unnecessarily collected onto a single worker and then sent to the client, whereas we could just have the client fetch separate partitions from separate workers (which may or may not happen all at once if it needs to transit through the scheduler).

This replicates the issue with the current finalizer methods in dask/dask, which are created by `dask.compute(df)` but are skipped by `df.compute()`.